pkg/vault/test/foundry/QueryERC4626Buffer.t.sol

```diff
@@ -7,100 +7,77 @@ import "forge-std/Test.sol";
 import { IERC4626 } from "@openzeppelin/contracts/interfaces/IERC4626.sol";
 import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 
-import { TokenConfig, TokenType } from "@balancer-labs/v3-interfaces/contracts/vault/VaultTypes.sol";
-import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
 import { IBatchRouter } from "@balancer-labs/v3-interfaces/contracts/vault/IBatchRouter.sol";
 import { IRateProvider } from "@balancer-labs/v3-interfaces/contracts/vault/IRateProvider.sol";
 
 import { ERC4626TestToken } from "@balancer-labs/v3-solidity-utils/contracts/test/ERC4626TestToken.sol";
 import { ArrayHelpers } from "@balancer-labs/v3-solidity-utils/contracts/test/ArrayHelpers.sol";
 
-import { PoolMock } from "../../contracts/test/PoolMock.sol";
+import { BaseERC4626BufferTest } from "./utils/BaseERC4626BufferTest.sol";
 
-import { BaseVaultTest } from "./utils/BaseVaultTest.sol";
-
-contract QueryERC4626BufferTest is BaseVaultTest {
+contract QueryERC4626BufferTest is BaseERC4626BufferTest {
     using ArrayHelpers for *;
 
-    ERC4626TestToken internal waDAI;
-    ERC4626TestToken internal waUSDC;
-
-    // For two-token pools with waDAI/waUSDC, keep track of sorted token order.
-    uint256 internal waDaiIdx;
-    uint256 internal waUsdcIdx;
-
-    address internal yieldBearingPool;
-
-    // The yield-bearing pool will have 100x the liquidity of the buffer.
-    uint256 internal yieldBearingPoolAmount = 10e6 * 1e18;
-    uint256 internal bufferAmount = yieldBearingPoolAmount / 100;
-    uint256 internal tooLargeSwapAmount = yieldBearingPoolAmount / 2;
+    uint256 internal tooLargeSwapAmount = erc4626PoolInitialAmount / 2;
     // We will swap with 10% of the buffer.
-    uint256 internal swapAmount = bufferAmount / 10;
+    uint256 internal swapAmount = bufferInitialAmount / 10;
 
     function setUp() public virtual override {
-        BaseVaultTest.setUp();
-
-        waDAI = new ERC4626TestToken(dai, "Wrapped aDAI", "waDAI", 18);
-        vm.label(address(waDAI), "waDAI");
-
-        // "USDC" is deliberately 18 decimals to test one thing at a time.
-        waUSDC = new ERC4626TestToken(usdc, "Wrapped aUSDC", "waUSDC", 18);
-        vm.label(address(waUSDC), "waUSDC");
-
-        (waDaiIdx, waUsdcIdx) = getSortedIndexes(address(waDAI), address(waUSDC));
-
-        _initializeBuffers();
-        _initializeYieldBearingPool();
+        bufferInitialAmount = erc4626PoolInitialAmount / 100;
```

Comment:
We're using `bufferInitialAmount` before this line to set the `swapAmount`. Shouldn't we be updating `swapAmount` after?

Reply:
Yes, you are right